src/librustc_target/abi/mod.rs

```diff
@@ -660,7 +660,11 @@ impl FieldPlacement {
 
     pub fn offset(&self, i: usize) -> Size {
         match *self {
-            FieldPlacement::Union(_) => Size::ZERO,
+            FieldPlacement::Union(count) => {
+                assert!(i < count,
+                        "Tried to access field {} of union with {} fields", i, count);
```

Comment:
We should eventually figure out an uniform style across the compiler for this sort of thing.

I'm personally for `// Foo bar.` for comments but `bug!("foo bar");` for ICEs.

(but you can leave this as-is)